eng/versioning.targets

```diff
@@ -22,20 +22,18 @@
     </AssemblyMetadata>
   </ItemGroup>
 
-  <!-- Adds SupportedOSPlatform attribute for Windows Specific libraries -->
-  <ItemGroup Condition="'$(IsWindowsSpecific)' == 'true' and '$(IsTestProject)' != 'true'">
-    <AssemblyAttribute Include="System.Runtime.Versioning.SupportedOSPlatform">
-        <_Parameter1>windows</_Parameter1>
-    </AssemblyAttribute>
-  </ItemGroup>
-
-  <ItemGroup>
+  <!-- Adds SupportedOSPlatform or UnsupportedOSPlatform attributes -->
+  <ItemGroup Condition="'$(IsTestProject)' != 'true'">
+    <_supportedOSPlatforms Include="$(SupportedOSPlatforms)" />
     <_unsupportedOSPlatforms Include="$(UnsupportedOSPlatforms)" />
   </ItemGroup>
 
-  <Target Name="AddUnsupportedOSPlatformAttribute" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild">
+  <Target Name="AddAssemblyOSPlatformAttributes" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild">
     <ItemGroup>
-      <AssemblyAttribute Include="System.Runtime.Versioning.UnsupportedOSPlatform" Condition="'@(_unsupportedOSPlatforms)' != ''">
+      <AssemblyAttribute Include="System.Runtime.Versioning.SupportedOSPlatform" Condition="'@(_supportedOSPlatforms)' != ''">
+        <_Parameter1>%(_supportedOSPlatforms.Identity)</_Parameter1>
+      </AssemblyAttribute>
+      <AssemblyAttribute Include="System.Runtime.Versioning.UnsupportedOSPlatform" Condition="'@(_supportedOSPlatforms)' == '' and '@(_unsupportedOSPlatforms)' != ''">
         <_Parameter1>%(_unsupportedOSPlatforms.Identity)</_Parameter1>
       </AssemblyAttribute>
```

Comment:
I think you can remove the second half of the condition.

Reply:
Yeah we don't need `_supportedOSPlatforms condition.

Reply:
Ah, good catch. Thanks.

Reply:
I was checking `_supportedOSPlatforms` here to prevent any `[UnsupportedOSPlatform]` attributes from occurring when there are already `[SupportedOSPlatform]` attributes, as we must only emit one or the other (with _Supported_ being more restrictive and therefore taking precedence).

I'll take the suggestion of emitting a build warning when this conflict occurs, but I recommend we keep the check in place as having both attributes in place can lead to lots of other noise in the compilation, including even crashes in the analyzer runner at the moment.

Reply:
Makes sense. However we should be able to catch these kind of issues on PR review? And the warning makes sense.